spec/lib/balancer_delivery_method_spec.rb

```diff
@@ -145,6 +153,57 @@ def rand(_)
     end
   end
 
+  context 'when the email does not bypass unverified mail protection' do
+    let(:mail) { ExampleMailer.greet(email, bypass_unverified_mail_protection:) }
+    let(:bypass_unverified_mail_protection) { false }
+
+    before do
+      ActionMailer::Base.balancer_settings = { mock_smtp: 10 }
+      mail.deliver_now
+    end
+
+    context 'when the email belongs to a user' do
+      let(:email) { user.email }
+      let(:user) { create(:user, email: 'u@a.com', email_verified_at:) }
+
+      context 'and the email is not verified' do
+        let(:email_verified_at) { nil }
+
+        it { expect(mail).not_to have_been_delivered_using(MockSmtp) }
```

Comment:
Je crois que le warning de codecov qui te dit que `User#unverified_email?` n'est pas testé révèle que ce test ne fais pas ce qu'on croit. J'ai l'impression que `user` n'est pas créé ici  car il n'est que dans un `let` et on l'appelle pas dans `it`

Reply:
Bien vu !

Le soucis n'était pas dans le `let` car le `it` appelle `mail` qui appelle `email` qui appelle `user` ...
Mais dans le message Grettings qui ne s'était pas le `to` comme je le supposais.

C'est fixé !